server.go

```diff
@@ -480,29 +442,94 @@ func (a *App) reply(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	// locking?
-	_, err = a.redis.LPush(a.ctx, "msgbus.system.reply", body).Result()
+	_, err = a.redis.LPush(a.ctx, "msgbus.system.reply", body)
 	if err != nil {
 		err = errors.Wrap(err, "couldn't push entry to reply queue")
 		w.Write(errorReply(err.Error()))
 	}
 	w.Write(successReply())
 
 }
+func (a *App) Serve(ctx context.Context) error {
+	a.ctx = ctx
+	go a.runServer(ctx)
+	var serverErr error
+	go func() {
+		if serverErr = a.server.ListenAndServe(); serverErr != http.ErrServerClosed {
+			log.Error().Err(serverErr).Msg("server ListenAndServe failed")
+		}
+	}()
+	interval := 2 * time.Second
+	timer := time.NewTimer(interval)
+	for {
```

Comment:
if this is just to wait for the ctx cancellation to stop the server you need to do 
```go
<- ctx.Done()
return a.server.Shutdown()
```
this will basically wait until ctx is canceled then once canceled calls server.shutdown.

> NOTE: if you read the docs for a.server.Shutdown() you will notice the ctx passed to it is used to set a "timeout" to wait on active connections to disconnect. so it doesn't make sense to pass same context to it (because it'a already canceled, instead u should create another context with timeout of like 5 or 10 seconds. and then call shutdown